src/measure_theory/l1_space.lean

```diff
@@ -44,79 +134,101 @@ assume hfi hgi,
       lintegral_nnnorm_add hfm hgm
     ... < ⊤ : add_lt_top.2 ⟨hfi, hgi⟩
 
--- We don't need `f` to be measurable here, but it's easier to have a uniform API
-@[nolint]
-lemma lintegral_nnnorm_neg {f : α → γ} (hf : measurable f) :
+lemma lintegral_nnnorm_neg {f : α → β} :
   (∫⁻ (a : α), ↑(nnnorm ((-f) a))) = ∫⁻ (a : α), ↑(nnnorm ((f) a)) :=
 lintegral_congr_ae $ by { filter_upwards [], simp }
 
-lemma integrable_neg {f : α → γ} (hfm : measurable f) : integrable f → integrable (-f) :=
-assume hfi, calc _ = _ : lintegral_nnnorm_neg hfm
+lemma integrable_neg {f : α → β} : integrable f → integrable (-f) :=
+assume hfi, calc _ = _ : lintegral_nnnorm_neg
                  ... < ⊤ : hfi
 
+lemma integrable_sub {f g : α → β} (hf : measurable f) (hg : measurable g) :
+  integrable f → integrable g → integrable (f - g) :=
+λ hfi hgi,
+  by { rw sub_eq_add_neg, refine integrable_add hf (measurable_neg hg) hfi (integrable_neg hgi) }
+
+lemma integrable_norm {f : α → β} (hfi : integrable f) : integrable (λa, ∥f a∥) :=
+calc (∫⁻ (a : α), (nnnorm ∥f a∥)) = (∫⁻ (a : α), (nnnorm (f a))) :
+    begin
+      apply lintegral_congr_ae, filter_upwards [],
+      assume a,
+      simp only [mem_set_of_eq],
+      rw [nnnorm_norm]
+    end
+  ... < ⊤ : hfi
+
 section normed_space
-variables {K : Type*} [normed_field K] [normed_space K γ]
+variables {K : Type*} [normed_field K] [normed_space K β]
 
-lemma integrable_smul {c : K} {f : α → γ} (hfm : measurable f) : integrable f → integrable (c • f) :=
+lemma integrable_smul {c : K} {f : α → β} (hfm : measurable f) :
```

Comment:
I still think that you should be able to remove the measurability assumption on this one.

Reply:
I think the reasoning is
```lean
∫ ∥(c • f) x∥ dx = ∫ ∥c∥ * ∥f x∥ dx = ∥c∥ * ∫ ∥f x∥ dx < ⊤
```
But somehow the second equality needs to use `measurable f`:
```lean
lemma lintegral_const_mul (r : ennreal) {f : α → ennreal} (hf : measurable f) :
  (∫⁻ a, r * f a) = r * (∫⁻ a, f a) :=
calc (∫⁻ a, r * f a) = (∫⁻ a, (⨆n, (const α r * eapprox f n) a)) :
    by congr; funext a; rw [← supr_eapprox_apply f hf, ennreal.mul_supr]; refl
  ... = (⨆n, r * (eapprox f n).integral) :
  begin
    rw [lintegral_supr],
    { congr, funext n, rw [← simple_func.const_mul_integral, ← simple_func.lintegral_eq_integral] },
    { assume n, exact simple_func.measurable _ },
    { assume i j h a, exact canonically_ordered_semiring.mul_le_mul (le_refl _)
        (monotone_eapprox _ h _) }
  end
  ... = r * (∫⁻ a, f a) : by rw [← ennreal.mul_supr, lintegral_eq_supr_eapprox_integral hf]
```
`measurable f` is used in the last line: `lintegral_eq_supr_eapprox_integral hf`.

Reply:
The result is still true without measurability, because `measurable f` can also be removed from `lintegral_const_mul`. You can just come back to the definition of integral, take a simple function `s` bounded by `f`, then `r s` is a simple function bounded by `rf `, and then taking the supremum of the integral over such `s` you get the inequality `r \int f <= \int r f`. The other inequality follows by applying this to `1/r` and `r f` (separating the cases `r =0` and `r = top` maybe). I think it is worth doing it.

Reply:
My mistake, this is wrong in `lintegral_const_mul`, for `r = top`: one can have a function `f` which is positive everywhere, but for which the only simple minorant is the function 0. Then the integral of `f` is 0, but the integral of `top * f` is `top`. 

However, this holds for `r` different from `top`, which is enough for `integrable_smul`.

Reply:
```lean

lemma lintegral_const_mul_le (r : ennreal) (f : α → ennreal) : r * (∫⁻ a, f a) ≤ (∫⁻ a, r * f a) :=
begin
 rw [lintegral, ennreal.mul_supr],
  refine supr_le (λs, _),
  rw [ennreal.mul_supr],
  simp only [supr_le_iff, ge_iff_le],
  assume hs,
  rw ← simple_func.const_mul_integral,
  refine le_supr_of_le (const α r * s) (le_supr_of_le (λx, _) (le_refl _)),
  exact canonically_ordered_semiring.mul_le_mul (le_refl _) (hs x)
end

lemma lintegral_const_mul' (r : ennreal) {f : α → ennreal} (hr : r ≠ ⊤) : 
  (∫⁻ a, r * f a) = r * (∫⁻ a, f a) :=
begin
  by_cases h : r = 0,
  { simp [h] },
  apply le_antisymm _ (lintegral_const_mul_le r f),
  have rinv : r * r⁻¹  = 1 := ennreal.mul_inv_cancel h hr,
  have rinv' : r ⁻¹ * r = 1, by { rw mul_comm, exact rinv },
  have := lintegral_const_mul_le (r⁻¹) (λx, r * f x),
  simp [(mul_assoc _ _ _).symm, rinv'] at this,
  simpa [(mul_assoc _ _ _).symm, rinv] 
    using canonically_ordered_semiring.mul_le_mul (le_refl r) this
end
```

Reply:
Oh, that's very nice of you. I really should do it myself...